### PR TITLE
Add esm version for legacy index

### PR DESCRIPTION
**Pull request by BPScott**
Don't add esm for other js files as I highly doubt anybody uses them, as
the index contains everything.

This allows for better tree-shaking and easier integration with
esm-based bundlers

Replace the content of index.common.js and index.d.ts with the output of
running tsc over this new esm file. This improves esm-cjs interop in
index.common.js, as it allows you to use named imports when importing
index.common.js from an .mjs file in nodejs.

The following now works when before it was a error in an mjs file:
`import {colorSkyDark} from './dist/index.common.js';`

---

ESM & CJS interop is hecking complex thanks to the myriad of tools that have overlapping subsets of support. Fixing this properly for both dist and dist-modern requires breaking changes that I don't really want to spend time deeply investigating right now. This is a decent halfway house.

Note that this doesn't effect dist-modern, which is still outputing cjs. Fixing that without weird hacks will involve either removing dist entirely, or swapping the entrypoints around so that the modern stuff is what you get when you import `@shopify/polaris-tokens` and you get at legacy stuff with a full path eg `@shopify/polaristokens/dist-legacy/index.esm.js`, which is a breaking change.

### CODE REVIEW

**Comment by BPScott** [comment]
Sticking this in draft while I do some testing

**Comment by kaelig** [comment]
Sounds good! Happy to do some testing in web or polaris-styleguide. Perhaps shipping an alpha would be helpful so we can easily look at how this looks in Shrink Ray reports?

Also: please add an item to the changelog.

File: index.js [hunk 1]
```diff
@@ -1 +1,5 @@
+// This file used to be the main entrpoint for the package, but now main points
+// directly to `dist/index.common.js`. This file is left in the unlikely event
+// that somebody importing `@shopify/polaris-tokens/index.js`.
```
Comment (BPScott) [inline index.js:3]:
```suggestion
// that somebody is importing `@shopify/polaris-tokens/index.js`.
```

**Comment by BPScott** [comment]
> Sounds good! Happy to do some testing in web or polaris-styleguide. Perhaps shipping an alpha would be helpful so we can easily look at how this looks in Shrink Ray reports?

I've tested this by modifying content in polaris-react's node_modules and after uncovering and fixing a a small gotcha everything looks good. We need an explict default export to not break backwards compat, even though 99% of references in web and polaris-react use named exports.

**Review by kaelig — APPROVED** [review]
Classic tokens (in `dist/`) work as expected!

Tested using the polaris-icons website, across both esm/commonjs and using these 3 import strategies: `import {xxx}`, `import tokens`, and `import * as tokens`.

**Pull request merged**